etna/analysis/plotters.py

```diff
@@ -1035,3 +1036,94 @@ def plot_imputation(
+        all the strategies can be examined at `etna.analysis.feature_selection.AggregationMode`
+    periodogram_params:
+        additional keyword arguments for periodogram, `scipy.signal.periodogram` is used
+    segments:
+        segments to use
+    columns_num:
+        if `relevance_aggregation_mode="per-segment"` number of columns in subplots, otherwise the value is ignored
+    figsize:
+        size of the figure per subplot with one segment in inches
+
+    Notes
+    -----
+    In non per-segment mode all segments are cut to be the same length, the last values are taken.
+
+    Raises
+    ------
+    ValueError:
+        if period < 2
+    ValueError:
+        if periodogram can't be calculated on segment because of the NaNs inside it
+    """
+    if period < 2:
+        raise ValueError("Period should be at least 2")
+    if periodogram_params is None:
+        periodogram_params = {}
+    if not segments:
+        segments = sorted(ts.segments)
+
+    df = ts.to_pandas()
+
+    # plot periodograms
+    if amplitude_aggregation_mode == "per-segment":
+        ax = prepare_axes(segments=segments, columns_num=columns_num, figsize=figsize)
+        for i, segment in enumerate(segments):
+            segment_df = df.loc[:, pd.IndexSlice[segment, "target"]]
+            segment_df = segment_df[segment_df.first_valid_index() : segment_df.last_valid_index()]
+            if segment_df.isna().any():
+                raise ValueError(f"Periodogram can't be calculated on segment with NaNs inside: {segment}")
+            frequencies, spectrum = periodogram(x=segment_df, fs=period, **periodogram_params)
+            ax[i].step(frequencies, spectrum)
+            ax[i].set_xscale("log")
+            ax[i].set_title(f"Periodogram: {segment}")
+    else:
+        # find length of each segment
+        lengths_segments = []
+        for segment in segments:
+            segment_df = df.loc[:, pd.IndexSlice[segment, "target"]]
+            segment_df = segment_df[segment_df.first_valid_index() : segment_df.last_valid_index()]
+            if segment_df.isna().any():
+                raise ValueError(f"Periodogram can't be calculated on segment with NaNs inside: {segment}")
+            lengths_segments.append(len(segment_df))
+        cut_length = min(lengths_segments)
+
+        # cut each segment to `cut_length` last elements and find periodogram for each segment
+        frequencies_segments = []
+        spectrums_segments = []
+        for segment in segments:
+            segment_df = df.loc[:, pd.IndexSlice[segment, "target"]]
+            segment_df = segment_df[segment_df.first_valid_index() : segment_df.last_valid_index()][-cut_length:]
+            frequencies, spectrum = periodogram(x=segment_df, fs=period, **periodogram_params)
+            frequencies_segments.append(frequencies)
+            spectrums_segments.append(spectrum)
+
+        frequencies = frequencies_segments[0]
```

Comment:
Why did we create frequencies_segments array if we need only one value from it?

Reply:
I think, that this implementation is easy. Other implementations that came to my mind require writing if cases for some iteration, or rewriting the same value inside the array, that can be considered as a bug. Current implementation looks very simple.
Do you know simple alternatives?